go/vt/vttablet/heartbeat/reader.go

```diff
@@ -80,7 +80,6 @@ func (r *Reader) Open(dbc dbconfigs.DBConfigs) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 	if r.isOpen {
-		log.Fatalf("BUG: Reader object cannot be initialized twice without closing in between: %v", r)
```

Comment:
LGTM
Sorry about coming in late on this one. Generally, sub-service Open and Close functions have to be idempotent. For example: https://github.com/youtube/vitess/blob/master/go/vt/vttablet/tabletserver/tx_engine.go#L90. The only guarantee given by TabletServer is that it won't call Open or Close when another Open or Close is currently in progress.

Even in the case of a `StateServing` -> `StateServing` transition, let's say `replica`->`rdonly`, the Open functions will all get called again on services that are already open.